projects/form-builder/src/lib/pipes/show-field/show-field.pipe.ts

```diff
@@ -1,20 +1,63 @@
 import {Pipe, PipeTransform} from '@angular/core';
+import {Observable, of} from 'rxjs';
+import {map, startWith} from 'rxjs/operators';
 import {CompiledField} from '../../interfaces/compiled-field.interface';
 import {Parser} from '../../utils/parser';
+import {safeEval} from '../../utils/safe-eval';
 
 @Pipe({
   name: 'showField'
 })
 export class ShowFieldPipe implements PipeTransform {
-  transform(fields: CompiledField[], parser: Parser): CompiledField[] {
+  transform(fields: CompiledField[], parser: Parser): Observable<CompiledField[]> {
+
+    return fields.some(field => field.condition)
+      ? parser.form.valueChanges.pipe(
+        startWith({}),
+        map(() => this.filtered(fields, parser))
+      )
+      : of(this.filtered(fields, parser));
+  }
```

Comment:
We need to build on this. Our observables should be defined based on the `deps` property. For instance when `deps: ['/email', '/name']`  we would build the observable that looks something like this: 

```
combineLatest(
  field.condition.deps.map(({pointer}) => parser.pointers[pointer].control.valueChanges)
)
  .pipe(map(values => this.checkCondition(values, field.condition.condition)))
```